src/main/java/de/tum/in/ase/parser/strategy/GCCParser.java

```diff
@@ -0,0 +1,154 @@
+package de.tum.in.ase.parser.strategy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import de.tum.in.ase.parser.domain.Issue;
+import de.tum.in.ase.parser.domain.Report;
+
+public class GCCParser implements ParserStrategy {
+
+    // Locations in regex group
+    protected static final int NO_SEGS = 3;
```

Comment:
What is `SEGS`? The full name would probably be better. Is `NO` used as in ‘not’, or as ‘number of’?
I also added few more comments about more declarative constant names for others.

Reply:
Is there a reason why the declarations are `protected` instead of `private`?

Reply:
> Is there a reason why the declarations are `protected` instead of `private`?

Initially, I wanted to inherit from [CheckstyleFormatParser](https://github.com/ls1intum/static-code-analysis-parser/blob/master/src/main/java/de/tum/in/ase/parser/strategy/CheckstyleFormatParser.java), but then didn't choose to do so. So yes, private should be fine at the moment. I could still change the implementation to extend CheckstyleFormatParser, but I think it should be fine either way.

Reply:
> What is `SEGS`? The full name would probably be better. Is `NO` used as in ‘not’, or as ‘number of’?
> I also added few more comments about more declarative constant names for others.

Segment refers to the segments that I divide the GCC output in, which is basically the first line that contains all the more general info, and then the part that contains the trace. `NO` was meant in the sense of "number as".  I am working on a refactor of the code right now, so then it hopefully should become more clear.